RGOPS-5199: Update Python SDK

akintunde-rocketgate opened the pull request:
- Updated Python SDK (GatewayService, GatewayRequest and GatewayResponse)
- Added GatewayCode
- Added more feature example code
- Enabled hostname lookup on LIVE transaction processing for maintenance window sensing
- Improved documentation

akintunde-rocketgate committed:
RGOPS-5199: Update Python SDK

Updated Python SDK and added more examples

volodymyr-rg commented on RocketGate.py:
```diff
@@ -164,109 +165,132 @@ class GatewayRequest:
+    # - 02 (390x400)
+    # - 03 (500x600)
+    # - 04 (600x400)
+    # - 05 (Full page)
```
I would advise turning this into docstring comment (https://stackoverflow.com/a/6061254). This way it improves IDE experience:
![image](https://github.com/rocketgate/rocketgate-python-sdk/assets/152193906/c7deb1d4-9281-46cc-8eb5-08df40c71ef7)

volodymyr-rg commented on RocketGate.py:
```diff
@@ -164,109 +165,132 @@ class GatewayRequest:
+    #	Set() - Set a value in the parameter list.
+    #
+    ######################################################################
+    #
```
The same way I would advise turning all such comments into docstrings according to Python conventions (please see https://peps.python.org/pep-0257/)

volodymyr-rg commented on RocketGate.py:
```diff
@@ -24,18 +24,17 @@
+    ######################################################################
+    #
     VERSION_INDICATOR = "version"
     VERSION_NUMBER = "PY3.5"
```
We need to increment this version number.

akintunde-rocketgate committed:
Merge branch 'main' into rgops-5199-update-python-sdk

akintunde-rocketgate committed:
Add more gateway related fields and codes

- Add Gateway response and request fields
- Remove unused Gateway response and request fields
- Add more GatewayCode entries
- Remove unused GatewayCode entries

akintunde-rocketgate committed:
Add more gateway related fields and codes

- Improve code documentation
- Add Gateway response and request fields
- Remove unused Gateway response and request fields
- Add more GatewayCode entries
- Remove unused GatewayCode entries

akintunde-rocketgate committed:
Bump up version number

3.5 -> 3.6

akintunde-rocketgate committed:
Review changes

- Rearrange fields alphabetically
- Update function documentation
- Removed redundant fields

akintunde-rocketgate committed:
Updated Documentation

Ensure docstring type is used

akintunde-rocketgate committed:
Resolve warnings

akintunde-rocketgate committed:
Revised connection object managemnt

akintunde-rocketgate committed:
Updated GatewayRequest and GatewayResponse fields

Used the documentation as the baseline for modification

akintunde-rocketgate committed:
Restored missing fields

akintunde-rocketgate committed:
Update: Align RocketGate.py with PHP SDK

akintunde-rocketgate committed:
Moved examples to /examples folder

akintunde-rocketgate committed:
Example code walkthrough for correctness

akintunde-rocketgate committed:
Added unit test cases

akintunde-rocketgate committed:
Tested example files

Included a module import statement to have script runnable from the commandline

akintunde-rocketgate committed:
Add more GatewayRequest & GatewayResponse fields not found in the PHP SDK

akintunde-rocketgate committed:
Reorder import statements

akintunde-rocketgate committed:
Updated REASON & RESPONSE codes for literals

akintunde-rocketgate committed:
Convert GatewayCodes to string

akintunde-rocketgate committed:
Remove int() casting and work with strings all through

akintunde-rocketgate committed:
Merge response_code and reason_code check for consistency

akintunde-rocketgate committed:
Add missing fields to test and example scripts

akintunde-rocketgate committed:
Added Shebang to files without it

akintunde-rocketgate committed:
Removed repeated variable definitions

akintunde-rocketgate committed:
Host set not lost in maintenance window setting

akintunde-rocketgate committed:
Use GATEWAY_URL in the PerformTransaction operation

akintunde-rocketgate committed:
Update ACHTest.py to extend BaseTestCase

akintunde-rocketgate committed:
Clear card detail from ACH request

akintunde-rocketgate committed:
Updated PerformLookup docs

akintunde-rocketgate committed:
Update example scrpts

petricarg committed:
Fix Set None case.
Sync with php examples and tests.

petricarg committed:
Fix TODOs and CR comments.

akintunde-rocketgate committed:
Merge pull request #2 from akintunde-rocketgate/rgops-5199-new-updates

Fix Set None case.
